engine/rpcserver.go

```diff
@@ -4175,8 +4176,171 @@ func (s *RPCServer) CurrencyStateTradingPair(_ context.Context, r *gctrpc.Curren
 		ai)
 }
 
+func (s *RPCServer) buildFuturePosition(position *order.Position, getFundingPayments, includeFundingRates, includeOrders, includePredictedRate bool) *gctrpc.FuturePosition {
```

Comment:
I do stuff like this all across the `engine` package to prevent it being used and modified in other contexts. Its a reused function that is only used for the RPCServer